modules/atlas-basic/variables.tf

```diff
@@ -0,0 +1,170 @@
+    description = "Atlas cluster number of read only nodes"
+    default     = 0
+    type        = number
+}
+variable "electable_nodes" {
+    description = "Atlas cluster number of electable nodes"
+    default     = 3
+    type        = number
+}
+
+variable "auto_scaling_disk_gb_enabled" {
+    description = "Atlas cluster auto scaling disk enabled"
+    default     = false
+    type        = bool
+}
+
+variable "disk_size_gb" {
+    description = "Atlas cluster disk size in GB"
+    default     = 10
+    type        = number
+}
+variable "provider_name" {
+  description = "Atlas cluster provider name"
+  default     = "AWS"
+  type        = string
+}
+variable "backing_provider_name" {
+  description = "Atlas cluster backing provider name"
+  default = "AWS"
+  type        = string
+}
+variable "provider_instance_size_name" {
+  description = "Atlas cluster provider instance name"
+  default     = "M10"
+  type        = string
+}
+
+variable "region" {
+    description = "Atlas cluster region"
+    default     = "US_EAST_1"
+    type        = string
+}
+variable "aws_region"{
+    description = "AWS region"
+    default     = "us-east-1"
+    type        = string
+}
+
+variable "mongo_version" {
+  description = "Atlas cluster version"
+  default     = "4.4"
+  type        = string
+}
+
+
+variable "user" {
+  description = "MongoDB Atlas User"
+  type        = list(string)
+  default     = ["dbuser1", "dbuser2"]
+}
+variable "password" {
+  description = "MongoDB Atlas User Password"
+  type        = list(string)
+}
+variable "database_name" {
+  description = "The Database in the cluster"
+  type        = list(string)
+}
+
+# database user
+variable "role_name" {
+    description = "Atlas database user role name"
+    default     = "readWrite"
+    type        = string
+}
+
+# IP Access List
+variable "cidr_block" {
+    description = "IP Access List CIDRs"
+    type        = list(string)
+    default     = ["10.1.0.0/16","12.2.0.0/16"]
+}
+
+variable "ip_address" {
+    description = "IP Access List IP Addresses"
+    type        = list(string)
+    default     = ["208.169.90.207","0.0.0.0"]
+}
+# aws
+
+variable "aws_vpc_cidr_block" {
+    description = "AWS VPC CIDR block"
+    default     = "10.0.0.0/16"
+    type        = string
+
+}
+
+variable "aws_route_table_cidr_block" {
+    description = "AWS route table CIDR block"
+    default     = "0.0.0.0/0"
+    type        = string
+}
+
+variable "aws_subnet_cidr_block1" {
+    description = "AWS subnet CIDR block"
+    default     = "10.0.1.0/24"
+    type        = string
+}
+variable "aws_subnet_cidr_block2" {
+  description = "AWS subnet CIDR block"
+  default     = "10.0.2.0/24"
+  type        = string
+}
+
+variable "aws_subnet_availability_zone1" {
+    description = "AWS subnet availability zone"
+    default     = "us-east-1a"
+    type        = string
+}
+variable "aws_subnet_availability_zone2" {
+  description = "AWS subnet availability zone"
+  default     = "us-east-1b"
```

Comment:
let;s use us-east-1 instead

Reply:
availability zones are supposed to be 'us-east-1a',  'us-east-1b', etc